jujugui/static/gui/src/app/state/state.js

```diff
@@ -0,0 +1,283 @@
+      }
+    });
+    return state;
+  }
+
+  /**
+    Parses the search portion of the url without the
+    State.PATH_DELIMETERS.get('search') key value.
+    @param {Array} urlParts - The url path split into parts.
+    @param {Object} state - The application state object as being parsed
+      from the URL.
+    @return {Object} The updated state to contain the search value, if any.
+  */
+  _parseSearch(urlParts, state) {
+    if (urlParts.length > 0) {
+      state.search = urlParts.join('/');
+    }
+    return state;
+  }
+
+  /**
+    Parses the GUI portion of the url without the
+    State.PATH_DELIMETERS.get('gui') key value.
+    @param {Array} urlParts - The url path split into parts.
+    @param {Object} state - The application state object as being parsed
+      from the URL.
+    @return {Object} The updated state to contain the search value, if any.
+  */
+  _parseGUI(urlParts, state) {
+    let indexes = [];
+    State.GUI_PATH_DELIMETERS.forEach(section => {
+      const index = urlParts.indexOf(section);
+      if (index > -1) {
+        indexes.push(index);
+      }
+    });
+    // If there were no sections found then just return.
+    if (!indexes.length) {
+      return state;
+    }
+    // JavaScript array sort sorts alphabetically which causes issues if
+    // you have multiple default sections ie) [0, 9, 17] gets sorted as
+    // [0, 17, 9]. So we must pass a custom sorter to it so that it sorts
+    // numerically.
+    indexes.sort((a, b) => a-b);
+    // Split out and store the sections
+    let guiParts = {};
+    indexes.forEach((index, arIndex) => {
+      const end = indexes[arIndex+1] || urlParts.length;
+      guiParts[urlParts[index]] = urlParts.slice(index+1, end).join('/');
+    });
+    state.gui = guiParts;
+    return state;
+  }
+
+  /**
+    Parses the url and extracts the user delimeted sections.
+    @param {Array} urlParts - The url path split into parts.
+    @param {Object} state - The application state object as being parsed
+      from the URL.
+    @return {Object}
+  */
+  _parseUser(urlParts, state) {
+    // Grab the indexes of the user delimeters as there can be multiple.
+    let indexes = [];
+    urlParts.forEach((item, index) => {
+      if (item === State.PATH_DELIMETERS.get('user')) {
+        indexes.push(index);
+      }
+    });
+    // If we have no user parts then we don't need to parse them.
+    if (indexes.length > 0) {
+      switch (indexes.length) {
+        case 1:
+          // Extract out the user portion of the list and then remove the
+          // user delimeter at the beginning.
+          const userBlock = urlParts.splice(indexes[0]).slice(1);
+          // The userBlock might have user components and store components.
+          // The user component has at most two spots. If it has more, then
+          // it may be a user store path.
+          if (userBlock.length > 2) {
+            // Check the 2 index spot to see if it's either an integer or a
```

Comment:
Check the second index, or 2nd index?
Also, "either a series or a revision", an integer seems to geenric in this context.